pkg/agent/externalnode/external_node_controller_linux_test.go

```diff
@@ -0,0 +1,1155 @@
+}
+
+func TestDeleteExternalNode(t *testing.T) {
+	_, addr1, _ := net.ParseCIDR("10.20.30.40")
+	_, addr2, _ := net.ParseCIDR("10.20.30.50")
+	iface := &net.Interface{
+		Index:        1,
+		HardwareAddr: net.HardwareAddr{0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88},
+		MTU:          1500,
+	}
+	iface2 := &net.Interface{
+		Index:        2,
+		HardwareAddr: net.HardwareAddr{0x33, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88},
+		MTU:          1500,
+	}
+	defer mockGetInterfaceConfig([]mockGetInterfaceConfigParam{
+		{iface, []*net.IPNet{addr1}, nil},
+		{iface2, []*net.IPNet{addr2}, nil},
+	})()
+	defer mockLinkByName(t, 2)()
+	defer mockRenameInterface(nil)()
+	defer mockHostInterfaceExists(true)()
+	defer mockConfigureLinkAddresses(nil)()
+	defer mockConfigureLinkRoutes(nil)()
+	c := newExternalNodeController(t)
+	mockIfaceStore.EXPECT().GetInterfacesByType(interfacestore.ExternalEntityInterface).Return([]*interfacestore.InterfaceConfig{
+		&if1,
+		&if2,
+	})
+	mockOFClient.EXPECT().UninstallVMUplinkFlows(if1.InterfaceName).Return(nil).Times(1)
+	mockOVSBridgeClient.EXPECT().DeletePort(if1.PortUUID).Return(nil).Times(1)
+	mockOVSBridgeClient.EXPECT().DeletePort(if1.UplinkPort.PortUUID).Return(nil).Times(1)
+	mockOVSCtlClient.EXPECT().DeleteDPInterface(if1.InterfaceName).Times(1)
+	mockIfaceStore.EXPECT().DeleteInterface(&if1).Times(1)
+	mockOFClient.EXPECT().UninstallVMUplinkFlows(if2.InterfaceName).Return(nil).Times(1)
+	mockOVSBridgeClient.EXPECT().DeletePort(if2.PortUUID).Return(nil).Times(1)
+	mockOVSBridgeClient.EXPECT().DeletePort(if2.UplinkPort.PortUUID).Return(nil).Times(1)
+	mockOVSCtlClient.EXPECT().DeleteDPInterface(if2.InterfaceName).Times(1)
+	mockIfaceStore.EXPECT().DeleteInterface(&if2).Times(1)
+	c.deleteExternalNode()
+	assert.Nil(t, c.syncedExternalNode)
+}
+
+func TestGetHostInterfaceName(t *testing.T) {
+	for _, tt := range []struct {
+		name                       string
+		iface                      v1alpha1.NetworkInterface
+		getIPNetDeviceFromIPParams []mockGetIPNetDeviceFromIPParam
+		expectedIfName             string
+		expectedIPs                []string
+		expectedError              error
+	}{
+		{
+			name:  "no device for IP",
+			iface: v1alpha1.NetworkInterface{IPs: []string{"1.1.1.3"}},
+			getIPNetDeviceFromIPParams: []mockGetIPNetDeviceFromIPParam{
+				{
+					link: &net.Interface{
+						Name:         "vEthernet (pod0-6631b7)",
+						HardwareAddr: net.HardwareAddr{0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88},
+						Index:        4,
+						Flags:        net.FlagUp,
+					},
+					getIPNetDeviceFromIPErr: nil,
+				},
+			},
+			expectedIfName: "vEthernet (pod0-6631b7)",
+			expectedIPs:    []string{"1.1.1.3"},
+			expectedError:  nil,
+		},
+		{
+			name:  "no device for IP",
+			iface: v1alpha1.NetworkInterface{IPs: []string{"1.1.1.3", "2408:8207:2561:ae11:cdee:f435:9904:c7dc"}},
+			getIPNetDeviceFromIPParams: []mockGetIPNetDeviceFromIPParam{
+				{
+					link: &net.Interface{
+						Name: "vEthernet (pod0-6631b7)",
```

Comment:
This interface name is following Windows vNIC name format, but this test file is focusing on Linux?

Reply:
updated